src/plugins/faultManagement/FaultManagementListView.vue

```diff
@@ -0,0 +1,333 @@
+                        .forEach(selectedFault => {
+                            this.openmct.faults.acknowledgeFault(selectedFault, data);
+                        });
+                })
+                .catch(() => {
+                    // Do nothing
+                });
+
+            this.selectedFaults = {};
+        },
+        async toggleShelveSelected(faults = Object.values(this.selectedFaults), shelveData = {}) {
+            const { shelved = true } = shelveData;
+            if (shelved) {
+                let title = faults.length > 1
+                    ? `Shelve ${faults.length} selected faults`
+                    : `Shelve fault: ${faults[0].id.name}`
+                ;
+
+                const formStructure = {
+                    title,
+                    sections: [
+                        {
+                            rows: [
+                                {
+                                    key: 'comment',
+                                    control: 'textarea',
+                                    name: 'Comment',
+                                    pattern: '\\S+',
+                                    required: false,
+                                    cssClass: 'l-input-lg',
+                                    value: ''
+                                },
+                                {
+                                    key: 'shelveDuration',
+                                    control: 'select',
+                                    name: 'Shelve Duration',
+                                    options: FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS,
+                                    required: false,
+                                    cssClass: 'l-input-lg',
+                                    value: FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS[0].value
+                                }
+                            ]
+                        }
+                    ],
+                    buttons: {
+                        submit: {
+                            label: 'Shelve'
+                        }
+                    }
+                };
+
+                let data;
+                try {
+                    data = await this.openmct.forms.showForm(formStructure);
+                } catch (e) {
+                    // Do nothing
+                    return;
+                }
+
+                shelveData.comment = data.comment || '';
+                shelveData.shelveDuration = data.shelveDuration !== undefined
+                    ? data.shelveDuration
+                    : FAULT_MANAGEMENT_SHELVE_DURATIONS_IN_MS[0].value;
+            } else {
+                shelveData = {
+                    shelved: false
+                };
+            }
+
+            Object.values(faults)
+                .forEach(selectedFault => {
+                    this.openmct.faults.shelveFault(selectedFault, shelveData);
+                });
+
+            this.selectedFaults = {};
```

Comment:
Should we set `this.selectedFaults = {}` after the `openmct.faults.shelveFault `promise resolves? Or is that not returning a promise?

Reply:
yeah I thought about it but we dont get any response back from shelve API so at the moment we cant do anything after response (eg: no condition check success vs fail).